Protecs/src/main/java/uk/ac/ucl/protecs/behaviours/Demography.java

```diff
@@ -0,0 +1,221 @@
+package uk.ac.ucl.protecs.behaviours;
+
+import java.util.Arrays;
+import java.util.List;
+
+import sim.engine.SimState;
+import sim.engine.Steppable;
+import uk.ac.ucl.protecs.objects.Household;
+import uk.ac.ucl.protecs.objects.Location;
+import uk.ac.ucl.protecs.objects.Person;
+import uk.ac.ucl.protecs.sim.Params;
+import uk.ac.ucl.protecs.sim.WorldBankCovid19Sim;
+
+public class Demography {
+	
+	public class Aging implements Steppable {
+
+		Person target;
+		int ticksUntilNextBirthday = 365;
+		
+		public Aging(Person p, int ticksPerDay) {
+			this.target = p;
+			this.ticksUntilNextBirthday = 365 * ticksPerDay;
+		}
+		
+		@Override
+		public void step(SimState arg0) {
+			target.updateAge();
+			arg0.schedule.scheduleOnce(arg0.schedule.getTime() + ticksUntilNextBirthday, this);
+		}
+		
+	}
+	
+	public class Mortality implements Steppable{
+		
+		Person target;
+		int ticksUntilNextMortalityCheck = 0;
+		int tickToCauseMortality = Integer.MAX_VALUE;
+		WorldBankCovid19Sim world;
+		public Mortality( Person p, int ticksPerDay, WorldBankCovid19Sim myWorld ) {
+			this.target = p;
+			this.ticksUntilNextMortalityCheck = myWorld.params.ticks_per_year;
+			this.world = myWorld;
+		} 
+		// steps taken
+		@Override
+		public void step(SimState arg0) {
+			if(this.target.isAlive()) {
+			if (this.tickToCauseMortality < Integer.MAX_VALUE) {
+				causeDeath();
+			}
+			else {
```

Comment:
purpose of else clause here unclear - can we confirm?

Reply:
Else clause is needed. This method causes death if a date has been selected, or creates a date of death in the else clause. I've updated this with comments to make it clearer. I have also realised that we could probably run this at the start of the simulation just once, rather than running it yearly. I will raise a separate issue for this to eventually get round to doing!